Issue #289: Document the most common use case (#481)
sentrysoftware/metricshub: metricshub-doc/src/site/markdown/usecases/process-command-line.md

Reviewer: Rename file to `process-windows.md`

keywords: process command lines, windows
description: How to monitor process command lines with MetricsHub

> # Monitoring a process command line on Windows

Reviewer: What is important here is not the "command line" part, but the process. We could totally monitor the process by specifying its name, and it would be almost the same thing. We could actually explain that in the use case.

> # Monitoring a process command line on Windows

Reviewer: ```suggestion
# Monitoring a process on Windows
```

> **MetricsHub** makes it easy to monitor a process command line. To set up monitoring, you only need to create a connector based on the [Windows - Processes (WMI)](../connectors/windowsprocess.html) connector and specify the process command line to be monitored in the `config/metricshub.yaml` configuration file.

Reviewer: ```suggestion
**MetricsHub** makes it easy to monitor a process through its name or command line. To set up monitoring, we only need to specify a dedicated instance of the [Windows - Processes (WMI)](../connectors/windowsprocess.html) connector in the `config/metricshub.yaml` configuration file. Under this instance, we specify the name or the command line of the process to be monitored.
```

> In this example, we created a `metricshubWindowsProcess` connector based on the `WindowsProcess` connector. This connector is be activated and monitor the `metricshub` process command line. 

Reviewer: ```suggestion
In this example, we specify a dedicated instance of the `WindowsProcess` connector (`mssqlProcess`) to monitor the process of Microsoft SQL Server, i.e. the process(es) whose command line contains `sqlservr.exe`.
```

  > Note: This page provides just one example of the data collection feature available in **MetricsHub**. For more information, refer to [Customize Data Collection](../configuration/configure-monitoring.md#customize-data-collection).

## Procedure

> To monitor the `metricshub` process command line:

Reviewer: ```suggestion
To monitor the Microsoft SQL Server process:
```

1. In the `config/metricshub.yaml` file, we configure the monitoring on a local Windows machine through `WMI`:

    ```yaml
    resources:
      localhost:
        attributes:
          host.name: localhost
          host.type: windows
        protocols:
          wmi:
            timeout: 120
    ```

> 2. We create the `metricshubWindowsProcess` connector based on the [Windows - Processes (WMI)](../connectors/windowsprocess.html) connector: 

Reviewer: ```suggestion
2. We specify the `mssqlProcess` dedicated instance of the [Windows - Processes (WMI)](../connectors/windowsprocess.html) connector:
```

    ```yaml
        additionalConnectors:

>           metricshubWindowsProcess:

Reviewer: ```suggestion
          mssqlProcess:
```

            uses: WindowsProcess # Connector used
            force: true # Connector is always activated
    ```

> 3.  We specify the process command line to be monitored:

Reviewer: ```suggestion
3.  We specify the regular expression that will match with the command line of Microsoft SQL Server processes (note how we escape the dot in `sqlservr.exe`):
```

    ```yaml
            variables:

>               matchCommand: metricshub

Reviewer: ```suggestion
              matchCommand: "sqlservr\\.exe"
```
